demos/js/messageMonitor.js

```diff
@@ -0,0 +1,161 @@
+            start: "%timestamp - %port.name - %message.type: (no content to display)",
+            stop: "%timestamp - %port.name - %message.type: (no content to display)",
+            sysex: "%timestamp - %port.name - %message.type: (contents not displayed)",
+            tuneRequest: "%port.name - %message.type: (no content to display)"
+        },
+
+        modelListeners: {
+            loggedMessages: {
+                funcName: "youme.demos.messageMonitor.logger.appendLastMessage",
+                args: ["{that}.container", "{that}.model.loggedMessages", "{that}.options.templates"] // container, loggedMessages
+            }
+        }
+    });
+
+    youme.demos.messageMonitor.logger.appendLastMessage = function (container, loggedMessages, templates) {
+        if (loggedMessages.length) {
+            var lastMessage = loggedMessages[loggedMessages.length - 1];
+            var template = templates[lastMessage.message.type] || templates.message;
+            var messageContent = fluid.stringTemplate(template, lastMessage);
+            var childElement = $("<div class='message-monitor-log-message'></div>");
+            childElement.text(messageContent);
+            container.prepend(childElement);
+        }
+        else {
+            container.html("");
+        }
+    };
+
+    fluid.defaults("youme.demos.messageMonitor", {
+        gradeNames: ["youme.templateRenderer"],
+        markup: {
+            container: "<div class='message-monitor-inner-container'><div class='message-monitor-inputs'></div><div class='message-monitor-log'></div><button class='message-monitor-reset-button'>Clear Log</button></div>"
+        },
+
+        selectors: {
+            inputs: ".message-monitor-inputs",
+            log: ".message-monitor-log",
+            reset: ".message-monitor-reset-button"
+        },
+
+        model: {
+            loggedMessages: []
+        },
+
+        invokers: {
+            resetLog: {
+                funcName: "youme.demos.messageMonitor.resetLog",
+                args: ["{that}"]
+            }
+        },
+
+        components: {
+            inputs: {
+                type: "youme.multiPortSelectorView.inputs",
+                container: "{that}.dom.inputs",
+                options: {
+                    model: {
+                        portSpecs: [{
+                            name: "Launchpad Pro .+ (MIDI|Standalone Port)"
+                        }]
+                    },
+                    components: {
+                        portConnector: {
+                            options: {
+                                dynamicComponents: {
+                                    connection: {
+                                        options: {
+                                            // Listen at this level so that we can determine which port/device sent the message.
+                                            listeners: {
+                                                "onMessage.log": {
```

Comment:
Looks like a great candidate for being parcelled up in a grade which is then sent with an options distribution

Reply:
Done, much cleaner!